File: NET/week 5/Week5/RacingSimulator/Bicycle.cs
public class Bicycle : Vehicle
{
    public int Acceleration { get; set; }

    public Bicycle(string model, string color, int speed, int acceleration)
        : base(model, color, speed)
    {
        Acceleration = acceleration;
    }

    public override void CalculatePosition()
    {
        Position = Speed + Acceleration / 2;
    }

    public override void Move()
    {
        Console.WriteLine($"The bicycle {Name} is moving at {Speed} km/h.");
    }
}

File: JAVSCRIPT/week 3/day 12/assignment 9/bicycle.js


import Vehicle from './vehicle.js';

class Bicycle extends Vehicle {
    constructor(name, color, speed, acceleration) {
        super(name, color, speed);
        this.acceleration = acceleration;
    }

    move() {
        this.speed += 0.5 * this.acceleration;
        return this.speed;
    }
    
}

export default Bicycle;